### PR TITLE
fix(auth): guards against creating a user for the system actor

### PR DESCRIPTION
**Pull request by aditya-radhakrishnan**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-service/auth-impl/src/main/java/com/datahub/authentication/user/NativeUserService.java [hunk 1]
```diff
@@ -44,7 +44,11 @@ public void createNativeUser(@Nonnull String userUrnString, @Nonnull String full
     Objects.requireNonNull(password, "password must not be null!");
     Objects.requireNonNull(authentication, "authentication must not be null!");
 
-    Urn userUrn = Urn.createFromString(userUrnString);
+    final Urn userUrn = Urn.createFromString(userUrnString);
+    if (userUrn.toString().equals(SYSTEM_ACTOR)) {
+      throw new RuntimeException("Cannot create a user for the system actor");
```
[the remark this answers is not in the document]
Comment (aditya-radhakrishnan) [reply metadata-service/auth-impl/src/main/java/com/datahub/authentication/user/NativeUserService.java:49]: We do, but the frontend can decide what exactly to display to the user. Imo this is just an edge case that's not covered bc the System Actor doesn't exist in the database.

[the remark this answers is not in the document]
Comment (joshuaeilers) [reply metadata-service/auth-impl/src/main/java/com/datahub/authentication/user/NativeUserService.java:49]: Gotcha. It looks like this is already a urn on the frontend already in a constants file.

**Review by joshuaeilers — APPROVED** [review]
Thanks for fixing this!

**Review by joshuaeilers — APPROVED** [review]

File: metadata-service/auth-impl/src/main/java/com/datahub/authentication/user/NativeUserService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply metadata-service/auth-impl/src/main/java/com/datahub/authentication/user/NativeUserService.java:49]: Yes but is the frontend actually doing this?

Why don't we just say: This account already exists?

[the remark this answers is not in the document]
Comment (aditya-radhakrishnan) [reply metadata-service/auth-impl/src/main/java/com/datahub/authentication/user/NativeUserService.java:49]: I guess we could just move it to the block below and make it an OR condition. Seems fine to me.

[the remark this answers is not in the document]
Comment (aditya-radhakrishnan) [reply metadata-service/auth-impl/src/main/java/com/datahub/authentication/user/NativeUserService.java]: Refactored!

**Pull request merged**